From 2fffb1ed86b67cbf0092b9805e9ec5623629e5ab Mon Sep 17 00:00:00 2001
From: Olli Helenius <liff@iki.fi>
Date: Sun, 18 Dec 2022 17:12:59 +0200
Subject: [PATCH] Always prefer `XDG_CONFIG_HOME`

---
 cmd/root.go | 16 ++--------------
 1 file changed, 2 insertions(+), 14 deletions(-)

diff --git a/cmd/root.go b/cmd/root.go
index fc3e43e..808bfe3 100644
--- a/cmd/root.go
+++ b/cmd/root.go
@@ -94,25 +94,13 @@ func Execute(ctx context.Context, version string) {
 }
 
 func configLocations() []string {
-	home, err := os.UserHomeDir()
+	configFileName, err := xdg.ConfigFile("act/rc")
 	if err != nil {
 		log.Fatal(err)
 	}
 
-	configFileName := ".actrc"
-
-	// reference: https://specifications.freedesktop.org/basedir-spec/latest/ar01s03.html
-	var actrcXdg string
-	for _, fileName := range []string{"act/actrc", configFileName} {
-		if foundConfig, err := xdg.SearchConfigFile(fileName); foundConfig != "" && err == nil {
-			actrcXdg = foundConfig
-			break
-		}
-	}
-
 	return []string{
-		filepath.Join(home, configFileName),
-		actrcXdg,
+		configFileName,
 		filepath.Join(".", configFileName),
 	}
 }
-- 
2.39.2

